server/sonar-web/src/main/js/apps/permissions/project/store/actions.js

```diff
@@ -0,0 +1,107 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as api from '../../../../api/permissions';
+import { parseError } from '../../../code/utils';
+import { raiseError } from '../../shared/store/actions';
+import { getQuery, getFilter } from '../../shared/store/rootReducer';
+
+export const loadHolders = projectKey => (dispatch, getState) => {
+  const query = getQuery(getState());
+  const filter = getFilter(getState());
+
+  dispatch({ type: 'REQUEST_HOLDERS', query });
+
+  const requests = [];
+
+  if (filter !== 'groups') {
+    requests.push(api.getPermissionsUsersForComponent(projectKey, query));
+  } else {
+    requests.push(Promise.resolve([]));
+  }
+
+  if (filter !== 'users') {
+    requests.push(api.getPermissionsGroupsForComponent(projectKey, query));
+  } else {
+    requests.push(Promise.resolve([]));
+  }
+
+  return Promise.all(requests).then(responses => (
+      dispatch({
+        type: 'RECEIVE_HOLDERS_SUCCESS',
+        users: responses[0],
+        groups: responses[1],
+        query
+      })
+  )).catch(e => {
+    return parseError(e).then(message => dispatch(raiseError(message)));
+  });
+};
+
+export const updateQuery = (projectKey, query = '') => dispatch => {
+  dispatch({ type: 'UPDATE_QUERY', query });
+  if (query.length === 0 || query.length > 2) {
+    dispatch(loadHolders(projectKey));
+  }
+};
+
+export const updateFilter = (projectKey, filter) => dispatch => {
+  dispatch({ type: 'UPDATE_FILTER', filter });
+  dispatch(loadHolders(projectKey));
+};
+
+export const grantToUser = (projectKey, login, permission) => dispatch => {
+  api.grantPermissionToUser(projectKey, login, permission).then(() => {
+    dispatch({ type: 'GRANT_PERMISSION_TO_USER', login, permission });
+  }).catch(e => {
+    return parseError(e).then(message => dispatch(raiseError(message)));
+  });
+};
+
+export const revokeFromUser = (projectKey, login, permission) => dispatch => {
+  api.revokePermissionFromUser(projectKey, login, permission).then(() => {
+    dispatch({ type: 'REVOKE_PERMISSION_TO_USER', login, permission });
+  }).catch(e => {
+    return parseError(e).then(message => dispatch(raiseError(message)));
+  });
+};
+
+export const grantToGroup = (projectKey, groupName, permission) => dispatch => {
+  api.grantPermissionToGroup(projectKey, groupName, permission).then(() => {
+    dispatch({
+      type: 'GRANT_PERMISSION_TO_GROUP',
+      groupName,
+      permission
+    });
+  }).catch(e => {
+    return parseError(e).then(message => dispatch(raiseError(message)));
+  });
+};
+
+export const revokeFromGroup = (projectKey, groupName, permission) => dispatch => {
+  api.revokePermissionFromGroup(projectKey, groupName, permission).then(() => {
+    dispatch({
+      type: 'REVOKE_PERMISSION_FROM_GROUP',
+      groupName,
+      permission
+    });
+  }).catch(e => {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Remove curly braces and "return" from this arrow function body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)